libtenzir/src/execution_node.cpp

```diff
@@ -834,6 +848,9 @@ auto exec_node(
     return exec_node_actor::behavior_type::make_empty_behavior();
   }
   self->state.weak_node = node;
+  self->attach_functor([name = self->state.op->name()] {
+    TENZIR_DEBUG("exec node for {} shuts down", name);
```

Comment:
This should be past tense, because the functor is executed by the clock _after_ the actor was shut down, i.e., it is not safe to access `*self` anymore.

```suggestion
    TENZIR_DEBUG("exec-node for {} shut down", name);
```